features/step_definitions/login.rb

```diff
@@ -0,0 +1,49 @@
+
+Given(/^I prepare the state of the app as logout user$/) do
+  set_current_user
+
+  case app_state
+  when :logout
+    # home page opens as logout by default
+  when :login
+    steps %{
+      When I go to my profile
+      When I log out
+    }
+  end
+end
+
+When(/^I go to my profile$/) do
+  @current_page = page(ProductsPage).await(timeout: 30)
+  @current_page.go_to_profile
+end
+
+When(/^I log out$/) do
+  @current_page = page(ProfilePage).await(timeout: 30)
+  @current_page.logout
+end
+
+When(/^I log in$/) do
+  pending # Write code here that turns the phrase above into concrete actions
+end
+
+When(/^I touch the log-in button$/) do
+  @current_page.touch_login_button
+end
+
+Given(/^I am in the login page$/) do
+  @current_page = page(LoginPage).await(timeout: 30)
+end
+
+When(/^I fill in "(.*?)" as "(.*?)"$/) do |input, text|
+  @current_page.fill_in_input(input, text)
```

Comment:
just a question :) does the convenience of not passing the page to some of the methods worth thinking and adding @current_page to any other method that change the page?

Reply:
I think @current_page should be recreated when the current page changes so some step definitions uses the defined @current_page.

Reply:
I understand the logic, my question was about the meaningfulness of this approach (maybe I miss smth and can use it in my project) :) 

From my point of view only thing that you gain by that `@current_page` is that you do not have to pass the page to each step (which is nice), but does it worth handling the current page change all over the places? (this is what I'm not sure about).

Reply:
My test would look like:
`When I click on FirstPage.settingsButton`
`Then I see SettingsScreen.logoutButton`
so I have to handle pages by passing them.

Reply:
With this way, you are adding logic to the business part (cucumber) which seems to be done in page.

Reply:
With this way, you are adding logic to the business part (cucumber) which seems to be done in page.